### ISSUE TITLE
LWFA simulation with mesh refinement and PSATD solver shows instability

### ISSUE THREAD

**Issue by prkkumar** [ECP-WarpX/WarpX#3390]
I noticed an instability when using PSATD solver with mesh refinement in an LWFA simulation. The following input file will reproduce the issue.
[input.txt](https://github.com/ECP-WarpX/WarpX/files/9552575/input.txt)

Instability is only on fine patch. Following plot shows Ez (note the high field values (on the scale of 1e55) on the fine patch) and corresponding charge density.
![Ez](https://user-images.githubusercontent.com/89051199/189782287-816c7b13-f7fa-4410-b205-7a6c25287d4d.png)
![rho](https://user-images.githubusercontent.com/89051199/189782299-2ca764b6-157d-44ef-916c-afac94cf2ef2.png)

Same configuration works with ckc solver.

Tests were done on https://github.com/RemiLehe/WarpX/tree/laser_deposit_on_main_grid branch which adds option to deposit laser only on the main grid (PR: https://github.com/ECP-WarpX/WarpX/pull/3235).

**Comment by prkkumar** [comment]
Running with `psatd.current_correction = 0` does not have any effect on the instability. Here is the Ez field with this option
![psatd_no_current_corr](https://user-images.githubusercontent.com/89051199/190518477-e57e2727-9f3d-4555-ac80-c31e6dfef352.png)

**Comment by ax3l** [comment]
Discussed today with @RemiLehe @EZoni :
- we could try to turn off PMLs around the MR patch to see if this causes the instability
- if it removes the instability, we know it's from the PMLs

**Comment by RemiLehe** [comment]
Hi @prkkumar 
Could you modify your local version of WarpX to replace these lines:
https://github.com/ECP-WarpX/WarpX/blob/development/Source/Initialization/WarpXInitData.cpp#L510
by
```
            do_pml_Lo[lev] = amrex::IntVect::TheZeroVector();
            do_pml_Hi[lev] = amrex::IntVect::TheZeroVector();
```
and the re-run the above simulation?

This would disable PMLs around the MR patch, and it would allow us to see whether the instability is caused by the PMLs.

**Comment by RemiLehe** [comment]
I actually just did the above test. The instability is still present, although it looks different without PML:

- **Without PML around the patch:**
![Screenshot from 2022-09-22 13-49-12](https://user-images.githubusercontent.com/6685781/191848201-c76e1b38-08af-4e3f-a9c3-5294ea2915af.png)


- **With PML around the patch:**
![Screenshot from 2022-09-22 13-47-57](https://user-images.githubusercontent.com/6685781/191848014-60c9b2ae-0f36-4436-9f66-77b4374bb322.png)

**Comment by RemiLehe** [comment]
I tried with `particles.deposit_on_main_grid = beam` and this does get rid of the instability:
![Screenshot from 2022-09-30 12-55-55](https://user-images.githubusercontent.com/6685781/193347617-87fbfd7a-4ac1-4250-ac89-528b3f7733f5.png)

**Comment by RemiLehe** [comment]
In addition, it seems that the instability is also removed when using `none` instead of `damped` for the longitudinal boundary. (Although the fields do not look quite right.) This is definitely a bug: the `damped` boundary condition should not have any influence on level 1.
![Screenshot from 2022-09-30 16-05-41](https://user-images.githubusercontent.com/6685781/193367625-ba9e6e2a-7a16-400f-9871-09b86a6f1baf.png)

**Comment by EZoni** [comment]
Note after offline discussion: the test reported here must be rerun after the bug fix in #3884 to check whether the issue still persists and/or if anything changed.

**Comment by prkkumar** [comment]
The bug fix in https://github.com/ECP-WarpX/WarpX/pull/3884 indeed gets rid of the instability. Results from a run with PSATD + MR + damped BC after the fix in #3884 is shown below  
![Screen Shot 2023-04-28 at 12 19 12 PM](https://user-images.githubusercontent.com/89051199/235234558-d26eb60b-ddb4-4a3d-8fce-02362ec6cd27.png)
